bee-common/bee-packable/src/packable/bounded_integer.rs

```diff
@@ -0,0 +1,130 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    coerce::CoerceInfallible,
+    error::{PackError, UnpackError},
+    packer::Packer,
+    unpacker::Unpacker,
+    Packable,
+};
+
+use core::convert::{Infallible, TryFrom, TryInto};
+
+/// Trait that provides an interface for bounded integers.
+pub trait Bounded {
```

Comment:
Should the trait be called `BoundedInteger` ? Bounded feels a bit too generic idk

Reply:
We can do, since that's really our only use-case. 

Theoretically though, this can be used with any type. In fact, I think we should probably add a `PartialOrd` constraint on the `Bounds` type to limit it to values that can be used in ranges, so the macros don't break if you try and use it with something else:
```rust
pub trait Bounded {
    /// The type used to define the bounds.
    type Bounds: PartialOrd;

    // ...
}
```
If we wanted to be really strict with it we could use `PrimInt` and `Unsigned` from `num_traits`, (and then BoundedInteger or BoundedUInteger would definitely fit) but that's an extra dependency we probably don't want.

Reply:
The file is called `bounded_integer` and the doc of the trait says `for bounded integers.` though :trollface:

Reply:
I don't mind it being more generic than just integer but then let's remove mentions of integers ? Other than the actual int impls ofc

Reply:
Yeah that makes sense. I've renamed the macro and made the definition more generic, but the files are still `bounded_integer`, since we're only implementing integers -- I can rename them if that's better though.

Reply:
Done :+1: